runtime/precompiles/src/NominationDAO.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// This is a PoC to use the staking precompile wrapper as a Solidity developer.
+
+pragma solidity >=0.8.0;
+
+import "./ParachainStakingWrapper.sol";
+
+contract NominationDao {
+    // TODO Our precompile should have an accessor for this.
+    uint256 constant MinNominatorStk = 5 ether;
+
+    /// The collator that this DAO is currently nominating
+    address public target;
+
+    /// The account that deployed this DAO. Currently this deployer chooses the nomination target
+    /// Once the basic PoC stuff is done we can add cooler actual DAO logic. But for now we want it simple
+    address public deployer;
+
+    /// The ParachainStaking wrapper at the known pre-compile address. This will be used to make all calls
+    /// to the underlying staking solution
+    ParachainStaking public precompile;
+
+    /// Whether this DAO currently has a nomination active
+    /// TODO Our precompile should have an accessor for this that wraps the pallet's in_nominator.
+    bool public isNominating;
+
+    //TODO Do we need this constructor at all, or can we initialize this stuff directly.
+    constructor() {
+        // Initialize the deployer and the
+        target = msg.sender;
+        deployer = msg.sender;
+
+        // Instantiate the parachain staking wrapper at the known precompile address (decimal 256)
+        precompile = ParachainStaking(
+            0x0000000000000000000000000000000000000100
+        );
```

Comment:
@sorpaas Could you please check this approach for errors. The PR description explains that we are using a precompile to wrap features of a specific pallet. This is similar to your dispatch precompile, but less generic.

The problem I'm facing is when I try to call from this solidity code into the precompile, the call goes to the wrong address. I've also tried passing in the precompile address as a parameter to the constructor.

The funny thing is when I submit calls directly to the precompile, they work as expected.

Reply:
Update: The problem with calling precompiles from solidity code was that the EVM will not make the inner call unless _some_ code exists at that contract address. I've solved / worked around this by deploying simple revert code under each precompile address.